### PR TITLE
fix(query): do not match queries unless all required args are present in the response object

### PR DESCRIPTION
**Pull request by littlemooon**
We were previously identifying a matching query for a model based on the returned type.

This is not valid in the case where `ArchivedResponse` is queried with different required args than can be returned on the response object.

The fix finds the matching response object and confirms that the fields are available to be returned, to then be passed into this matching query at a later date, from the model instance.

### CODE REVIEW

File: packages/codegen-doc/src/query.ts [hunk 1]
```diff
@@ -18,17 +19,42 @@ export function findQuery(
...
+  const responseFieldNames = context.objects
+    .find(obj => reduceTypeName(queryMatchingType.type) === obj.name.value)
+    ?.fields?.map(responseField => responseField.name.value);
+
+  if (!responseFieldNames) {
+    return undefined;
+  }
 
-  return matchedNameAndType ?? matchedType;
+  /** Ensure the fields are available on the object */
```
Comment (littlemooon) [inline packages/codegen-doc/src/query.ts:51]: The main change is contained to this file.

We were previously identifying a matching query for a model based on the returned type. 

This is not valid in the case where `ArchivedResponse` is queried with different required args than can be returned on the response object.

The fix finds the matching response object and confirms that the fields are available to be returned, to then be passed into this matching query at a later date, on the model.

File: packages/codegen-doc/src/query.ts [hunk 2]
```diff
@@ -18,17 +19,32 @@ export function findQuery(
...
 
-  return matchedNameAndType ?? matchedType;
+  /** Get the matching object definition */
+  const responseObject = context.objects.find(obj => type === obj.name.value);
+  const responseFieldNames = responseObject?.fields?.map(responseField => responseField.name.value);
+  if (!responseFieldNames?.length) {
+    return undefined;
+  }
+
+  /** Find a query with required args available on the response object */
```
Comment (littlemooon) [inline packages/codegen-doc/src/query.ts:45]: Main changes are contained to this file

File: packages/codegen-doc/src/query.ts [hunk 3]
```diff
@@ -18,17 +19,24 @@ export function findQuery(
...
 
-  const matchedNameAndType = context.queries.find(query => {
+  /** Get the matching object definition */
+  const responseObject = context.objects.find(obj => type === obj.name.value);
+  const responseFieldNames = responseObject?.fields?.map(responseField => responseField.name.value);
+  if (!responseFieldNames?.length) {
+    return undefined;
+  }
+
+  /** Get all queries matching type and have required args available on the response object */
```
Comment (littlemooon) [inline packages/codegen-doc/src/query.ts:29]: main change

**Review by eldh — APPROVED** [review]

**Comment by eldh** [comment]
Sweeeet

**Pull request merged**